src/pluggy/_async.py

```diff
@@ -0,0 +1,198 @@
+"""
+Async support for pluggy using greenlets.
+
+This module provides async functionality for pluggy, allowing hook implementations
+to return awaitable objects that are automatically awaited when running in an
+async context.
+"""
+
```

Comment:
@simonw i'd appreciate early feedback on this wrt utility for datasette

its not the deep integration like I wanted/hoped to implement at first

however its a directly working integration that doesn’t break on color mixing - but it does have the caveat of breaking never officieally supported threading patterns some people use

i currently intend to clean up history and then have this land and release